app/models/host.rb

```diff
@@ -1692,14 +1693,15 @@ def read_only_storages
     end
   end
 
-  def archived?
-    ems_id.nil?
+  def archived
+    has_attribute?("archived") ? self["archived"] : ems_id.nil?
```

Comment:
When would a host have or not have the archived attribute?  Unclear why this changed from being just `ems_id.nil?`

Reply:
If we use virtual attributes and `select(:id, :archived)`, then we will have the attribute `archived` but will not have the attribute `ems_id`.

This case happens in reports where we only bring back the columns that will be displayed in the report.

Reply:
Also, going to rails 6.1, they are a little more picky for the fields they bring back when there is a `select()`, so this may have creeped up in that migration